src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -802,21 +812,127 @@ internal Hover GetHoverItem(TextDocumentPosition textDocumentPosition, ScriptFil
+            {
+                try
+                {
+                    // Queue the task with the binding queue    
+                    QueueItem queueItem = this.BindingQueue.QueueBindingOperation(
+                        key: scriptParseInfo.ConnectionKey,
+                        bindingTimeout: LanguageService.PeekDefinitionTimeout,
+                        bindOperation: (bindingContext, cancelToken) =>
+                        {
+                            // Get suggestions for the token
+                            IEnumerable<Declaration> declarationItems;
+                            int parserLine = textDocumentPosition.Position.Line + 1;
+                            int parserColumn = textDocumentPosition.Position.Character + 1;
+                            declarationItems = Resolver.FindCompletions(
+                                scriptParseInfo.ParseResult, 
+                                parserLine, parserColumn, 
+                                bindingContext.MetadataDisplayInfoProvider);
+
+                            // Match token with the suggestions(declaration items) returned
+                            DeclarationType type = 0;
+                            string schemaName;
+                            PeekDefinition pd = new PeekDefinition(connInfo);
+                            foreach (Declaration declarationItem in declarationItems)
+                            {
+                                if (declarationItem.Title.Equals(tokenText))
+                                {
+                                    type = declarationItem.Type;
+                                    // Script object using SMO based on type
+                                    switch (type)
+                                    {
+                                        case DeclarationType.Table:
+                                            return pd.GetTableDefinition(tokenText);
+                                        case DeclarationType.View:
+                                            schemaName = this.GetSchemaName(scriptParseInfo, textDocumentPosition.Position, scriptFile);
+                                            return pd.GetViewDefinition(tokenText, schemaName);
+                                        case DeclarationType.StoredProcedure:
+                                            schemaName = this.GetSchemaName(scriptParseInfo, textDocumentPosition.Position, scriptFile);
+                                            return pd.GetStoredProcedureDefinition(tokenText, schemaName);
+                                        default:
+                                            return null;
+                                    }
+                                }
+                            }
+                            return null;
+
+                        },
+                        timeoutOperation: (bindingContext) =>
+                        {
+                            return null;
```

Comment:
I'd caution that, at least from my perspective, not doing anything at the UI isn't very intuitive. It suggests to me that the extension is broken or incapable of doing what it should. But, that would be a discussion for the extension